sdcm/nemesis.py

```diff
@@ -687,6 +798,141 @@ def _modify_table_property(self, name, val, filter_out_table_with_counter=False,
+        added_columns_info.get('column_names', None)
+        if not added_columns_info['column_names']:
+            remove = False
+        CMD = "ALTER TABLE " + self._add_remove_column_target_table[1]
+        if remove:
+            remove = []
+            for n in range(random.randrange(1,min(
+                    len(added_columns_info['column_names']), self._add_remove_column_max_per_remove
+            ))):
+                while True:
+                    column_name = random.choice(added_columns_info['column_names'])
+                    if column_name not in remove:
+                        break
+                remove.append(column_name)
+            CMD += " DROP " + (', '.join(remove))
+
+        if add:
+            add = []
+            for n in range(random.randrange(1, self._add_remove_column_max_per_add)):
+                while True:
+                    added_columns_info = self._get_added_columns_info(self._get_target_noncompact_table(
+                        self._add_remove_column_target_table), self._add_remove_column_columns_info)
+                    new_column_name = self._random_column_name(added_columns_info['column_names'].keys(),
+                                                               self._add_remove_column_max_column_name_size)
+                    new_column_type = FieldTypeDefinition.get_random(added_columns_info['column_types'], max_level=3)
+                    if new_column_type is None:
+                        break
+                    add.append([new_column_name, new_column_type])
+            CMD += " ADD %s" % (', '.join(["%s <%s>" % (col[0], col[1]) for col in add]))
+        self.log.debug("Add/Remove Column Nemesis: resulted CMD - %s" % (CMD,))
+        failed = False
+        try:
+            with self.tester.cql_connection_patient(self.target_node, keyspace=self._add_remove_column_target_table[0]) as session:
+                session.execute(CMD)
+        except Exception as e:
+            self.log.debug("Add/Remove Column Nemesis: SQL query execution has failed with error %s" % (str(e),))
+            failed = True
+        if not failed:
+            for column_name, column_type in add:
+                added_columns_info['column_names'][column_name] = column_type
+                column_type.remember_variant(added_columns_info['column_types'])
+
+    def modify_table_add_remove_column(self):
+        self.log.debug("Add/Remove Column Nemesis: Started")
+        self._add_remove_column_target_table = self._get_target_noncompact_table(self._add_remove_column_target_table)
+        if self._add_remove_column_target_table is None:
+            self.log.debug("Add/Remove Column Nemesis: No non-compact table was found. Creating new one.")
+            self._add_remove_column_target_table = ['add_remove_column_ks', 'add_remove_column_table']
+            self.log.debug("Add/Remove Column Nemesis: Table Created")
```

Comment:
This line says table created, but creation take place in the end of this function. why don't we do that here ? And actually do what the nemesis says on the cover ? I.e. if no compat table the nemsis should only crate the table and end